io-engine/src/core/snapshot.rs

```diff
@@ -0,0 +1,74 @@
+use async_trait::async_trait;
+
+/// Snapshot Captures all the Snapshot information for Lvol.
+#[derive(Debug)]
+pub struct SnapshotParams {
+    entity_id: Option<String>,
+    parent_id: Option<String>,
+    txn_id: Option<String>,
+    snap_name: Option<String>,
+}
+/// Implement Snapshot Common Function.
+impl SnapshotParams {
+    pub fn new(
+        entity_id: Option<String>,
+        parent_id: Option<String>,
+        txn_id: Option<String>,
+        snap_name: Option<String>,
+    ) -> SnapshotParams {
+        SnapshotParams {
+            entity_id,
+            parent_id,
+            txn_id,
+            snap_name,
+        }
+    }
+}
+
+///  Traits gives the common snapshot/clone interface for Local/Remote Lvol.
+#[async_trait(?Send)]
+pub trait SnapshotOps {
+    type Error;
+    /// Create Snapshot Common API.
+    async fn create_snapshot(
+        &self,
+        snap_param: SnapshotParams,
```

Comment:
Shouldn't creation itself require specific values for the id's? 
Seems like this api allows all of the none, which seems a bit odd.